echidna/limit/chi_squared.py

```diff
@@ -147,20 +124,23 @@ def pearson_chi_squared(observed, expected):
         events
 
     Raises:
-      ValueError: If either array contains a bin with content <= 0.0
+      ValueError: If arrays are different lengths.
 
     Returns:
       float: Calculated Pearson's chi squared
     """
-    correct_bin_content, message = check_bin_content(observed)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: observed " +
-                         message)
-    correct_bin_content, message = check_bin_content(expected)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: expected " +
-                         message)
-    return numpy.sum((observed-expected)**2 / expected)
+    if len(observed) != len(expected):
+        raise ValueError("Arrays are different lengths")
+    epsilon = 1e-34  # In the limit of zero
+    total = 0
+    for i in range(len(observed)):
```

Comment:
I think it's fine to leave this with the for loop for now, but it's just something we may want to think about improving in the future.